clientdb/order.go

```diff
@@ -689,6 +698,29 @@ func deserializeOrderTlvData(r io.Reader, o order.Order) error {
 		o.Details().ChannelType = order.ChannelType(channelType)
 	}
 
+	if allowedNodeIDsBytes, ok := parsedTypes[allowedNodeIDsType]; ok {
```

Comment:
This isn't really correct. You only get the `allowedNodeIDsBytes != nil` when the TLV library considers this to be an "unknown record type". So instead you should add a primitive record to the top that just expects a byte slice. Then here you can just check `t, ok := parsedTypes[allowedNodeIDSType]; ok && t == nil` and expect that byte slice to be filled with the data you want.